UnityProject/Assets/Scripts/Managers/GameManager.cs

```diff
@@ -754,18 +761,33 @@ public void RestartRound()
 		StartCoroutine(ServerRoundRestart());
 	}
 
-	IEnumerator ServerRoundRestart()
+	private float GetMemeoryUsagePrecentage()
 	{
-		Logger.Log("Server restarting round now.", Category.Round);
-		Chat.AddGameWideSystemMsgToChat("<b>The round is now restarting...</b>");
+		return (Profiler.GetTotalAllocatedMemoryLong() / 1048576) / SystemInfo.systemMemorySize * 100;
+	}
 
-		// Notify all clients that the round has ended
-		EventManager.Broadcast(Event.RoundEnded, true);
+	IEnumerator ServerRoundRestart()
+	{
+		string[] args = Environment.GetCommandLineArgs();
+		if ((ServerShutsDownOnRoundEnd == false || args.Contains("-NoReboot"))
+		    && (ServerAverageFPS >= 45 || GetMemeoryUsagePrecentage() <= 75f) || args.Contains("-AlwaysReboot") == false)
+		{
+			Logger.Log("Server restarting round now.", Category.Round);
+			Chat.AddGameWideSystemMsgToChat("<b>The round is now restarting...</b>");
+			// Notify all clients that the round has ended
+			EventManager.Broadcast(Event.RoundEnded, true);
 
-		yield return WaitFor.Seconds(0.2f);
+			yield return WaitFor.Seconds(0.2f);
 
-		CustomNetworkManager.Instance.ServerChangeScene("OnlineScene");
+			CustomNetworkManager.Instance.ServerChangeScene("OnlineScene");
 
-		StopAllCoroutines();
+			StopAllCoroutines();
+			yield break;
+		}
+		Logger.Log("Server is rebooting now. If you don't have a way to automatically restart the " +
```

Comment:
make this a log error so we can see it on Discord

Reply:
Done